crypto/bn256/gnark/g1_aff.go

```diff
@@ -0,0 +1,73 @@
+package bn256
+
+import (
+	"github.com/consensys/gnark-crypto/ecc/bn254/fp"
+)
+
+// This is just the addition formula
+// but given we know that we do not need Jacobian
+// coordinates, we use the naive implementation.
+//
+// Ideally, we push this into gnark
```

Comment:
As mentioned, I can remove this file and or upstream to gnark first -- added it as I noticed that most cryptography libraries are not well suited to the usecase where we do an operation on a point and then serialize straight after, most libraries asssume that you are going to make further computations on the result

Reply:
I did some rough benchmarks and it looked like this smooths out the variance between the worse cases